### PR TITLE
some suggestions

### PR DESCRIPTION
**Pull request by bryanwood**

### CODE REVIEW

File: Pages/Index.cshtml [hunk 1]
```diff
@@ -3,7 +3,7 @@
 <h1>From Razor</h1>
 
 <div id="app">
-    <router-view></router-view>
```
Comment (bryanwood) [inline Pages/Index.cshtml:6]: See `app/main.ts`

File: app/main.ts [hunk 2]
```diff
@@ -17,5 +17,6 @@ new Vue({
     router: new VueRouter({
         mode: 'history',
         routes: routes
-    })
+    }),
+    render: h => h('router-view')
```
Comment (bryanwood) [inline app/main.ts:21]: Use render function for top level so we don't need the template compiler at all.

File: webpack.config.js [hunk 3]
```diff
@@ -17,17 +17,14 @@ module.exports =  {
...
-        alias: { //this is needed to get template-compiler?
-            'vue': 'vue/dist/vue.common.js',
-        }
+        extensions: ['.vue', '.ts', '.js']
     },
     module: {
         rules: [
             {
                 test: /\.ts$/,
-                exclude: /node_modules|vue\/src/,
```
Comment (bryanwood) [inline webpack.config.js:29]: just exclude `node_modules`, anything else won't be imported as a dependency anyway.

File: webpack.config.js [hunk 4]
```diff
@@ -36,13 +33,14 @@ module.exports =  {
...
+                use: 'vue-loader'
             }
         ]
-    }
+    },
+    plugins: [
+        new webpack.optimize.CommonsChunkPlugin({
+            name: "vendor",
+            minChunks: Infinity
+        })
```
Comment (bryanwood) [inline webpack.config.js:44]: We can spit out any common imports into the vendor chunk as well (could make this a separate common chunk too).

File: app/types.d.ts [hunk 5]
```diff
@@ -0,0 +1 @@
+declare module "*.vue";
```
Comment (bryanwood) [inline app/types.d.ts:1]: Makes `import App from './App.vue'` acceptable to TSC

File: app/main.ts [hunk 6]
```diff
@@ -1,5 +1,5 @@
-import AppComponent from './app';
-import Bar from './app';
+import AppComponent from './app.vue';
+import Bar from './app.vue';
```
Comment (bryanwood) [inline app/main.ts:2]: See `app/types.d.ts`

File: VueFire.csproj [hunk 7]
```diff
@@ -2,6 +2,7 @@
 
   <PropertyGroup>
     <TargetFramework>netcoreapp2.0</TargetFramework>
+    <TypeScriptCompileBlocked>true</TypeScriptCompileBlocked>
```
Comment (bryanwood) [inline VueFire.csproj:5]: Prevent VS from trying to compile `.ts` files on save.

File: webpack.config.js (hunk 4 above)
Comment (ctolkien) [reply webpack.config.js:44]: Been looking at the way the VS templates do this. They have a seperate config for vendor bits, then use the https://webpack.js.org/plugins/dll-plugin/ to reference this in their main bundle. Which `provide means to split bundles in a way that can drastically improve build time performance.`

**Pull request merged**